### PR TITLE
feat(edgeless): fit duplicated content into view

### LINKED ISSUE #4426: Duplicated content in edgeless should fit into view
As in Miro, when the duplicated copy is in the field of view, no action is taken, and when the Duplicated copy is not in the field of view, it is scaled to place it in the field of view.

https://github.com/toeverything/blocksuite/assets/22443345/f20a864f-ed28-4699-961a-1482762b1ddc

### PR DESCRIPTION
**Pull request by WBbug**
close #4426

### CODE REVIEW

**Review by doodlewind — COMMENTED** [review]
I didn't see the effect here. Anything wrong with my operation?

https://github.com/toeverything/blocksuite/assets/7312949/6365b006-20fe-448b-a8fe-e7e2e775a01d

**Comment by WBbug** [comment]
> I didn't see the effect here. Anything wrong with my operation?
> 
>  Screen.Recording.2023-09-01.at.22.25.23.mov

Sorry for not dealing with the note, i just checked, there is an error about duplicated note. #4543 
maybe this problem is fixed I will updating the pr
image

**Comment by regischen** [comment]
maybe this method [setViewport](https://github.com/toeverything/blocksuite/blob/master/packages/blocks/src/surface-block/renderer.ts#L238)  can help you

**Comment by WBbug** [comment]
> maybe this method [setViewportByBound](https://github.com/toeverything/blocksuite/blob/master/packages/blocks/src/surface-block/renderer.ts#L238) can help you

thanks for help, i just update the pr

File: packages/blocks/src/page-block/edgeless/utils/clipboard-utils.ts [hunk 1]
```diff
@@ -61,10 +62,39 @@ export async function duplicate(
...
+  const { surface, page } = edgeless;
+  const { viewport } = surface;
+  const newElements = Array.from(
+    newElementIds,
+    id => surface.pickById(id) ?? page.getBlockById(id)
+  );
+  let totalBound = edgelessElementsBound(newElements as EdgelessElement[]);
+  totalBound = inflateBound(totalBound, 30);
+  let currentViewBound = Bound.from(viewport.viewportBounds);
+  while (!currentViewBound.contains(totalBound)) {
```
Comment (regischen) [inline packages/blocks/src/page-block/edgeless/utils/clipboard-utils.ts:88]:
```
why so complex computation.
totalBound.unite(currentViewBound) can work?
```

**Review by regischen — APPROVED** [review]

**Pull request merged**